anaUltraLatency.py

```diff
@@ -9,322 +9,48 @@
-        grVFATNSignalNoBkg.GetXaxis().SetRangeUser(-0.5,24.5)
-        grVFATNSignalNoBkg.Draw("APE1")
-        canv_Signal.SaveAs(filename+'/SignalNoBkg.png')
-
-    # Store - Sum over all VFATs
-    canv_LatSum = r.TCanvas("canv_LatSumOverAllVFATs","canv_LatSumOverAllVFATs",600,600)
-    canv_LatSum.cd()
-    hHitsVsLat_AllVFATs.SetXTitle("Latency")
-    hHitsVsLat_AllVFATs.SetYTitle("N")
-    hHitsVsLat_AllVFATs.GetXaxis().SetRangeUser(latMin,latMax)
-    hHitsVsLat_AllVFATs.Draw("hist")
-    canv_LatSum.SaveAs(filename + '/LatSumOverAllVFATs.png')
-
-    # Store - Max Hits By Lat Per VFAT
-    canv_MaxHitsPerLatByVFAT = r.TCanvas("canv_MaxHitsPerLatByVFAT","canv_MaxHitsPerLatByVFAT",1200,600)
-    canv_MaxHitsPerLatByVFAT.Divide(2,1)
-    canv_MaxHitsPerLatByVFAT.cd(1)
-    grNMaxLatBinByVFAT.SetTitle("")
-    grNMaxLatBinByVFAT.SetMarkerStyle(21)
-    grNMaxLatBinByVFAT.SetMarkerSize(0.7)
-    grNMaxLatBinByVFAT.SetLineWidth(2)
-    grNMaxLatBinByVFAT.GetXaxis().SetRangeUser(-0.5,24.5)
-    grNMaxLatBinByVFAT.GetXaxis().SetTitle("VFAT Pos")
-    grNMaxLatBinByVFAT.GetYaxis().SetRangeUser(0,nTrig)
-    grNMaxLatBinByVFAT.GetYaxis().SetTitle("Hit Count of Max Lat Bin")
-    grNMaxLatBinByVFAT.GetYaxis().SetTitleOffset(1.7)
-    grNMaxLatBinByVFAT.Draw("APE1")
-    canv_MaxHitsPerLatByVFAT.cd(2)
-    grMaxLatBinByVFAT.SetTitle("")
-    grMaxLatBinByVFAT.SetMarkerStyle(21)
-    grMaxLatBinByVFAT.SetMarkerSize(0.7)
-    grMaxLatBinByVFAT.SetLineWidth(2)
-    grMaxLatBinByVFAT.GetXaxis().SetTitle("VFAT Pos")
-    grMaxLatBinByVFAT.GetYaxis().SetTitle("Max Lat Bin")
-    grMaxLatBinByVFAT.GetYaxis().SetTitleOffset(1.2)
-    grMaxLatBinByVFAT.GetXaxis().SetRangeUser(-0.5,24.5)
-    grMaxLatBinByVFAT.Draw("APE1")
-    canv_MaxHitsPerLatByVFAT.SaveAs(filename+'/MaxHitsPerLatByVFAT.png')
-
-    # Store - TObjects
-    outF.cd()
-    hHitsVsLat_AllVFATs.Write()
-    grNMaxLatBinByVFAT.SetName("grNMaxLatBinByVFAT")
-    grNMaxLatBinByVFAT.Write()
-    grMaxLatBinByVFAT.SetName("grMaxLatBinByVFAT")
-    grMaxLatBinByVFAT.Write()
-    if options.performFit:
-        grVFATSigOverBkg.SetName("grVFATSigOverBkg")
-        grVFATSigOverBkg.Write()
-        grVFATNSignalNoBkg.SetName("grVFATNSignalNoBkg")
-        grVFATNSignalNoBkg.Write()
-    myT.Write()
-    outF.Close()
+    try:
+        args.func(
+                infilename      = args.infilename,
+                debug           = args.debug,
+                latSigMaskRange = args.latSigMaskRange,
+                latSigRange     = args.latSigRange,
+                outputDir       = filePath,
+                outfilename     = args.outfilename,
+                performFit      = args.performFit)
+    except IndexError as err:
+        printRed(err.message)
+        runCommand(["chmod", "-R", "g+rw", "{0}".format(filePath)])
+        exit(os.EX_SOFTWARE)
+    except IOError as err:
+        printRed(err.message)
+        runCommand(["chmod", "-R", "g+rw", "{0}".format(filePath)])
```

Comment:
> independently of whether the chmod action is the right one, could (should?) put this repeated command inside a finally so as to not repeat it

Yes so I was debating about putting this in a `finally` block.  But I thought that since you are calling inside the `except` block `exit` or `sys.exit` the `finally` block will not be executed right?  Or have I misunderstood "finally" in this case.

> lastly, probably minor (but possibly big enough to warrant a separate issue), but as you're pushing to have these done the "right" way(TM), should switch from exit to sys.exit (see python2 and python3)

This is a great point; and probably needs a general "Python2 vs. Python3" issue to address outstanding differences.

Reply:
`finally` will always run (`sys.exit` raises `SystemExit`)

Reply:
> `finally` will always run (`sys.exit` raises `SystemExit`)

Okay; so in this case I can apply this change you've suggested :+1: